StarWars/StarWarsOOP.cs

```diff
@@ -0,0 +1,135 @@
+        Station deathStar = new Station ("Death Star", "Imperial", 2);
+
+        //place people onto ships
+        falcon.EnterShip(leia,0);
+        falcon.EnterShip(hansolo,1);
+        tie.EnterShip(darth, 0);
+
+        //place ships onto stations
+        class1.EnterStation(falcon,0);
+        deathStar.EnterStation(tie,0);
+
+        //roll call for each station
+        // class1.roster(falcon);
+        Console.WriteLine(class1.roster(falcon));
+        Console.WriteLine(deathStar.roster(tie));
+    }
+}
+//creates a person with a first & last name, their alliance
+class Person {
+    private string firstName;
+    private string lastName;
+    private string alliance;
+    public Person (string firstName, string lastName, string alliance) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.alliance = alliance;
+    }
+
+    public string FullName {
+        get {
+            return this.firstName + " " + this.lastName;
+        }
+
+        set {
+            string[] names = value.Split (' ');
+            this.firstName = names[0];
+            this.lastName = names[1];
+        }
+    }
+}
+//creates ships that can hold people, Type and Alliance can be changed.
+//Passenger allows you to see who is onboard the ship
+class Ship {
+    public string name;
+    public Person[] passengers;
+    public Ship (string name, string alliance, string type, int size) {
+        this.name = name;
+        this.Type = type;
+        this.Alliance = alliance;
+        this.passengers = new Person[size];
+    }
+
+    public string Type {
+        get;
+        set;
+    }
+
+    public string Alliance {
+        get;
+        set;
+    }
+
+    public string Passengers {
+        get {
+            String formatted = " ";
+            foreach (var person in passengers) {
+                formatted += person + "\n";
+            }
+            return formatted;
+        }
+    }
+
+    public void EnterShip (Person person, int seat) {
+        this.passengers[seat] = person;
+    }
+
+    public void ExitShip (int seat) {
+        this.passengers[seat] = null;
+    }
+}
+//creates a station with a station name, alliance and port size
+//you can enter a station, leave a station- upon entering or leaving you need to update the roster.
+class Station {
+    public string StationName;
+    private Ship[] port;
+    public Station (string StationName, string alliance, int spot) {
+        this.StationName = StationName;
+        this.Alliance = alliance;
+        this.port = new Ship[spot];
+    }
+    public string station {
+        get;
+        set;
+    }
+    public string Alliance {
+        get;
+        set;
+    }
+    public void EnterStation (Ship name, int spot) {
+        this.port[spot] = name;
+    }
+
+    public void ExitStation (int spot) {
+        this.port[spot] = null;
+    }
+    //roster will have the passengers by ship enterStation
+    public string roster (Ship name) {
```

Comment:
What would i do if i wanted to print the rooster for all the ships on a space station?